src/databricks/labs/ucx/workspace_access/manager.py

```diff
@@ -144,6 +148,51 @@ def apply_group_permissions(self, migration_state: MigrationState) -> bool:
         logger.info("Permissions were applied")
         return True
 
+    def apply_group_permissions_experimental(self, migration_state: MigrationState) -> bool:
```

Comment:
```suggestion
    def apply_group_permissions_experimental(self, permission_migration: PermissionMigrationAPI, migration migration_state: MigrationState) -> bool:
```
this would reduce the diff size and remove the need to change so many tests.  unlike the current change in constructor, which is not strictly necessary then.

Reply:
we still need `self._ws.get_workspace_id()`, so would it make sense to change this to
```py
def apply_group_permissions_experimental(self, ws: WorkspaceClient, 
                                         migration_state: MigrationState) -> bool:
```